java/src/main/scala/org/typelevel/otel4s/java/trace/SpanBuilderImpl.scala

```diff
@@ -0,0 +1,237 @@
+
+      case None =>
+        Resource.pure(Span.Auto.fromBackend(Span.noopBackend))
+    }
+
+  def createRes[A](resource: Resource[F, A]): Resource[F, Span.Res[F, A]] = {
+    def child(name: String, ctx: JContext) =
+      createAutoSpan(
+        jTracer,
+        scope,
+        ctx,
+        jTracer.spanBuilder(name).setParent(ctx)
+      )
+
+    Resource.eval(parentContext).flatMap {
+      case Some(parent) =>
+        for {
+          span <- createAutoSpan(jTracer, scope, parent, makeJBuilder(parent))
+
+          result <- Resource.make(
+            child("acquire", span.ctx).surround(resource.allocated)
+          ) { case (_, release) =>
+            child("release", span.ctx).surround(release)
+          }
+
+          useSpan <- child("use", span.ctx)
+        } yield Span.Res.fromBackend(result._1, useSpan.backend)
+
+      case None =>
+        resource.map(a => Span.Res.fromBackend(a, Span.noopBackend))
+    }
+  }
+
+  private def makeJBuilder(parent: JContext): JSpanBuilder = {
+    val b = jTracer
+      .spanBuilder(name)
+      .setAllAttributes(Conversions.toJAttributes(attributes))
+      .setParent(parent)
+
+    kind.foreach(k => b.setSpanKind(toJSpanKind(k)))
+    b.setAllAttributes(Conversions.toJAttributes(attributes))
+    startTimestamp.foreach(d => b.setStartTimestamp(d.length, d.unit))
+    links.foreach { case (ctx, attributes) =>
+      b.addLink(toJSpanContext(ctx), Conversions.toJAttributes(attributes))
+    }
+
+    b
+  }
+
+  private def parentContext: F[Option[JContext]] =
+    parent match {
+      case Parent.Root =>
+        scope.root.map(s => Some(s.ctx))
+      case Parent.Auto =>
+        scope.current.map {
+          case TraceScope.Scope.Root(ctx)    => Some(ctx)
+          case TraceScope.Scope.Span(ctx, _) => Some(ctx)
+          case TraceScope.Scope.Noop         => None
+        }
+      case Parent.Explicit(parent) =>
+        scope.current.map {
+          case TraceScope.Scope.Root(ctx)    => Some(ctx.`with`(parent))
+          case TraceScope.Scope.Span(ctx, _) => Some(ctx.`with`(parent))
+          case TraceScope.Scope.Noop         => None
+        }
+    }
+}
+
+object SpanBuilderImpl {
+
+  sealed trait Parent
+
+  object Parent {
+    case object Auto extends Parent
+    case object Root extends Parent
+    final case class Explicit(parent: JSpan) extends Parent
+  }
+
+  private def createAutoSpan[F[_]: Sync](
+      jTracer: JTracer,
+      scope: TraceScope[F],
+      parent: JContext,
+      builder: JSpanBuilder
+  ): Resource[F, AutoSpanImpl[F]] = {
+
+    def acquire: F[SpanBackendImpl[F]] =
+      for {
+        now <- Sync[F].realTime
+        jSpan <- Sync[F].delay {
+          builder
+            .setStartTimestamp(now.length, now.unit)
+            .startSpan()
+        }
+      } yield new SpanBackendImpl(jTracer, jSpan, scope)
+
+    def reportStatus(backend: Span.Backend[F], ec: Resource.ExitCase): F[Unit] =
```

Comment:
It would be nice to make it configurable. i.e. https://github.com/trace4cats/trace4cats/blob/master/modules/kernel/src/main/scala/trace4cats/kernel/ErrorHandler.scala